### DISCUSSION TITLE
need help for first install. what am I doing wrong?

### DISCUSSION THREAD

**Discussion by lebeck69** [jomjol/AI-on-the-edge-device#1496] [Q&A]
hardware I have : 

- Ai-thinker esp32-cam mounted on MB.
- 32Gb Sandisk : formated in FAT32 (1 partition only)

I've done : 

1. get 13.0.4 version from (https://github.com/jomjol/AI-on-the-edge-device/releases)
2. copy folder 'sd-card' on the Sandisk with Wlan updated
3. web install : https://jomjol.github.io/AI-on-the-edge-device
3b. and later tryed flash_download_tool_3.9.3 with ..\AI-on-the-edge-device-13.0.4\releases\download

I've use chrome and firefox

Problems :

1. while on 'create reference' page' the 'LED intensity' value doesn't seem to be taken into account. The process still works and the reference is created tho. 
2a. on 'define alignment'  page, original images are broken. 
in edit_alignment.html I see 		
should it point to those ?
/config/ref0_org.jpg
/config/ref1_org.jpg
and not 
"/img_tmp/ref_zw_org.jpg"
"/img_tmp/ref_zw.jpg"

2b.Also when I'm trying to updatereference , it seems it's not writing on SD. I have 400 error : file already exist.

3. is that normal that HTML drop down menu for digits > model doesn't show any choices ? (same for analog)

I'm still able to finish Initial Setup but doesn't seem proper and esp32 seem to crash soon after booting.

4. in 'run time information'  last restart have some data... all the rest is blank! 

5. I just relized that I have this text on top .. over the menu
Digitizer - AI on the edge - 0 HTTP/1.1 500 Internal Server Error Content-Type: text/html Content-Length: 28 Failed to read existing fileHTTP/1.1 500 Internal Server Error Content-Type: text/html Transfer-Encoding: chunked 0

Any clue what I might have done wrong ?
Thx a lot already for taking the time to read.

I don't see much in the logs. they look like this : 
2022-12-05T23:59:23: =============================================================================================
2022-12-05T23:59:23: =================================== Main Started ============================================
2022-12-05T23:59:23: =============================================================================================
2022-12-05T23:59:23: Branch: 'HEAD', Tag: 'v12.0.1', Revision: dfe2c46, Date/Time: 2022-09-29 17:13

**Comment by jomjol** [comment]
Can you send a serial log?

**Comment by pillmd** [comment]
Servus!

Facing the same issues. I am also a first time installer. I was able to push through the setup, and now the esp32 seems to be doing something (taking pictures and storing them to sd card) but then the "overview" page shows just black screen with the ROI´s on top of it. 

I can confirm all of the problems stated above:
problem 1: same thing with the led intensity, i was just not sure if its my tired eye that cannot spot the difference.

problem 2: same issue, also visible in the logs attached

problem 3: I checked the config.ini file when I first copied it to sd card and it had 
`"Model = /config/dig-cont_0600_s3.tflite"`
and 
`"Model = /config/ana-cont_11.3.0_s2.tflite"` 
for the digits and analog dials. But after the initial configuration was done, the values were wiped out and the drop down in the GUI was blank (ofcourse with giving an error). You get the error right when you enter the config screen for the first time
unfortunately not visible in the logs since was done after disconnected from serial.

problem 4:System/info is also blank except "last restart". 

I also get "refused to connect" errors for the content update all the time.  

I was able to do serial logs when first started, unfortunately i didnot finish the installation since i wanted to connect It downstairs...I hope those can help and I hope the topicstarted would not mind if I attach them here.

[esp-web-tools-logs.txt](https://github.com/jomjol/AI-on-the-edge-device/files/10170593/esp-web-tools-logs.txt)

Cheers!

**Comment by lebeck69** [comment]
I feel less alone :)
thx pillmd

Big thx also to big boss jomjol for his work ad time ;)
Can't wait to make this work .. looks great

I can see in this log that it tries to reboot all the time.
I've seen other post about this.. gonna search and update this asap ..

[log_2022-12-07.txt](https://github.com/jomjol/AI-on-the-edge-device/files/10178313/log_2022-12-07.txt)

**Comment by jomjol** [comment]
This looks like something during the writing / saving of the config.ini went wrong. Can you repeat the setup procedure and set all needed entries?

**Comment by lebeck69** [comment]
just downloaded last version : 
https://github.com/jomjol/AI-on-the-edge-device/releases/tag/v13.0.5

copy C:\...\AI-on-the-edge-device-13.0.5\sd-card on the SD & update wlan.ini

![ai_on_the _edge_install](https://user-images.githubusercontent.com/116950228/206262130-65d0cc0f-0b5f-4e05-aba0-ae8f87a662e9.jpg)

I deco and reco the esp32. 
First 'Create Reference out of Raw Image' goes well exept LEDIntensity that doesn't seem to have effect (visually).
I save. 
reboot to activate changes then this happens on 'Define Alignment Structure in Reference Image' when clicking 'save'

![ai_on_the _edge_pbm](https://user-images.githubusercontent.com/116950228/206262142-57326c1c-6d2c-4747-a36c-1ab3b0be5633.jpg)

I've tryed leaving default value and update as on pictures  'Storage Path/Name:'
says 'Config.ini is updated!' straight after

by the way I think there is a small typo ...
'Mark the reference by drag and dop' > 'Mark the reference by drag and d**R**op ?!

I'm using following browser 
Chrome :Version 108.0.5359.99 (Build officiel) (64 bits)
Firefox 107.0.1 (64 bits)

**Comment by pillmd** [comment]
ok, I got mine to work ! the problem was most probably in the version 12 that was pushed through the installer. I tried first with OTA to upgrade to ver 13, which did not really work, so I made a clean install with the newer version, and new SD card. then everything worked as expected.

**Comment by lebeck69** [comment]
A- bought a other SD card (transcend 16gb). copyed all 13.0.5 files. updated Wlan.ini.
clean all chrome cache memory.

> same problem.
I relised that when i have the 400 error, it's cleaning /config/*.jpeg

B- manually create jpg with paint, put them in config folder, updated config.ini manually.
rebooted esp and continued configuration.

>not solving problem... even worst after restart, esp is loop rebooting every few minutes

Question : 

- Is my install procedure (described above) correct ?
- Do I need to install any java comptonment ?
- is it normal to enter flashdownload with ESP32 (while written ESP32-S on the board) ?

![flashdownload](https://user-images.githubusercontent.com/116950228/206552483-3527c306-4ddc-4dd8-bf10-3763e80ea11e.jpg)

I ordered a new esp32-cam to check that my first one was not broken.

**Reply by pillmd** [reply]
Hi @lebeck69 , i did not use the esp32 flashed, rather the "official" flasher (https://jomjol.github.io/AI-on-the-edge-device/index.html).
As I told earlier, first I tried to do an OTA update to 13.0.5 which gave me some of the items working (like the info about the model and the version) but i still got the ref images missing and still a lot of weird errors in the log. 

I did a clean flash then using the abovementioned tool, then formatted and prepared a sdcard with the data from "sdcard" folder. Booted the device and then did the config as required. I also did not reboot while doing the config, just save -> next .... save->next .. then rebooted at the very end. You mentioned a reboot during setting the alignment marks in your previous comments.
After the clean flash - all looks good, I also dont have connection issues anymore that I was referring to ("refuse to connect" errors). 

The only error I am currently getting is a new one  - "The Version of the Web Interface (?) does not match the Firmware Version (e23c89a)! It is suggested to keep them on the same version! " which I ignore at the moment. :)

good luck!

P.S. offtop  I just realized you have a magnet right there on the top, so why the hassle with the camera ? :D

**Comment by lebeck69** [comment]
wahhhhhhhhhhhh they've updated the web version
indeed seem to work way better now !
thx

had still message but doesn't seem a problem.
![Sans titre](https://user-images.githubusercontent.com/116950228/206671778-f43981db-90e4-4c63-8b9a-961cdc264f27.jpg)

even MQTT went straight away :D

Fine tuning to have the right numbers especially on red numbers but I'm exited


Thx for you reply
and again Jomjol and others.. for your impressive work

Any chance of building a discord or a closer integration with HA ?
